lib/network/modules/components/nodes/util/CircleImageBase.js

```diff
@@ -10,10 +10,44 @@ class CircleImageBase extends NodeBase {
 
   setOptions(options, imageObj, imageObjAlt) {
     this.options = options;
-    this.setImages(imageObj, imageObjAlt);
+
+    //
+    // The issue here is that this setImages() is called twice on initialization:
+    //
+    // - First here, via the constructor call in parent NodeBase
+    // - Second in the constructor in derived Image or CircleImage.
+    //
+    // As an extra complication, this setOptions() is also called
+    // on an update to the node, where the two images are passed
+    // for Image and CircleImage (only).
+    //
+    // In the first case, imageObj and ImageObjAlt are always undefined;
+    // we use this to differentiate the calls.
+    // TODO: this is hacky, find a better solution
+    //
+    if (!(imageObj === undefined && imageObjAlt === undefined)) {
+      this.setImages(imageObj, imageObjAlt);
+    }
   }
 
+
+  /**
+   * Set the images for this node.
+   *
+   * The images can be updated after the initial setting of options;
+   * therefore, this method needs to be reentrant. 
+   *
+   * For correct working in error cases, it is necessary to properly set
+   * field 'nodes.brokenImage' in the options.
+   *
+   * @param {Image} imageObj  required; main image to show for this node
+   * @param {Image|undefined} optional; image to show when node is selected
+   */
   setImages(imageObj, imageObjAlt) {
+    if (imageObj === undefined) {
```

Comment:
It *can* occur as user error, but the user will be notified through a check elsewhere. See response to your first review comment. In any case, `setImages()` is known to be broken and #3053 fixes it.

> Is there a standard in Visjs to separate things that users might see as...

Not that I know of, actually. Perhaps it's a good idea to differentiate the different kinds of errors. Something like prefix 'Internal error:' perhaps?